Meta/check-newlines-at-eof.py

```diff
@@ -0,0 +1,73 @@
+#!/bin/python3
+
+import os
+import subprocess
+import sys
+
+try:
+    os.chdir(os.path.dirname(__file__) + "/..")
+except:
+    os.exit(1)
+
+files = subprocess.run(
+    [
+        "git", "ls-files", "--",
+        "*.cpp",
+        "*.h",
+        "*.gml",
+        "*.html",
+        "*.js",
+        "*.css",
+        "*.sh",
+        "*.py",
+        ":!:Base",
+        ":!:Kernel/FileSystem/ext2_fs.h",
+        ":!:Libraries/LibC/getopt.cpp",
+        ":!:Libraries/LibCore/puff.h",
+        ":!:Libraries/LibCore/puff.cpp",
+        ":!:Libraries/LibELF/exec_elf.h"
+    ],
+    capture_output=True
+).stdout.decode().split('\n')
+
+MISSING_NEWLINE_AT_EOF_ERRORS = []
+MORE_THAN_ONE_NEWLINE_AT_EOF_ERRORS = []
+
+did_fail = False
+for filename in files:
+    try:
+        f = open(filename, "r")
+    except:
+        continue
```

Comment:
Same here, silently skipping files probably isn't a good idea.

Reply:
That's for the final element of files which is an empty line; I could also use `filename in files[:-1]` and remove the try block.

Reply:
That'd be a lot better, it's not obvious (without a comment). `stdout.decode().trim().split('\n')` should avoid this altogether and only put files in `files` :)

Reply:
I get this error:
```
Traceback (most recent call last):
  File "/home/emanuele6/pro/serenityOS/serenity/Meta/check-newlines-at-eof.py", line 9, in <module>
    files = subprocess.run(
AttributeError: 'str' object has no attribute 'trim'
```

Reply:
Oh, I think that method is called `strip`

Reply:
Ugh, it's trim in JS but strip in Python, I always get those mixed.

Reply:
Is there a method to only trim one `\n` from the end of the string? We probably don't want to remove spaces from the first and the last filename (just in case)

Reply:
rstrip with argument:

```python
>>> "   aaa   \n\n".rstrip("\n")
'   aaa   '
```

Reply:
`).stdout.decode()[:-1].split('\n')` we just use. but that will remove any character from the end which is not expressive.

Reply:
@linusg oh, I didn't see your comment.

You are right, I was worrying about using `rstrip('\n')` because the last filename could end in a `\n`, but that is already and issue since we are using `split('\n')`. let's just use `rstrip('\n')`. :laughing: